File: pcd/examples/VocSdkExample/VocSdkExample/VideoCell.m
//
//  VideoCell.m
//  VocSdkExample
//
//  Created by Yadhu Manoharan on 7/6/17.
//  Copyright © 2017 Akamai. All rights reserved.
//

#import <VocSdk/VocSdk.h>
#import "VideoCell.h"

@implementation VideoCell

- (void)awakeFromNib {
	// Initialization code
	[super awakeFromNib];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
	[super setSelected:selected animated:animated];

	// Configure the view for the selected state
}

- (void)setItem:(nonnull id<VocItem>)item
{
	self.title.text = item.title;
	if (item.thumbnail) {
		self.thumbnail.image = [UIImage imageWithContentsOfFile:item.thumbnail.localPath];
	}

	self.info2.text = [NSString stringWithFormat:@"contentID: %@",item.contentId];

	[self updateProgress:item.size bytesDownloaded:item.bytesDownloaded];

	switch (item.state) {

		case VOCItemDiscovered:
			self.backgroundColor = [UIColor brownColor];
			self.info.text = @"DISCOVERED";
			break;

		case VOCItemIdle:
			self.backgroundColor = [UIColor orangeColor];
			self.info.text = @"IDLE";
			break;

		case VOCItemQueued:
			self.backgroundColor = [UIColor whiteColor];
			self.info.text = @"QUEUED";
			break;

		case VOCItemPaused:
			self.backgroundColor = [UIColor yellowColor];
			self.info.text = @"PAUSED";
			break;

		case VOCItemDownloading:
			self.backgroundColor = [UIColor whiteColor];
			self.info.text = @"DOWNLOADING";
			break;

		case VOCItemCached:
			self.backgroundColor = [UIColor greenColor];
			self.info.text = @"CACHED";
			self.progress.hidden = YES;
			break;

		case VOCItemPartiallyCached:
			self.backgroundColor = [UIColor greenColor];
			self.info.text = @"P-CACHED";
			self.progress.hidden = YES;
			break;

		case VOCItemFailed:
			self.backgroundColor = [UIColor redColor];
			self.info.text = @"FAILED";
			break;

		default:
			self.backgroundColor = [UIColor grayColor];
			break;
	}

}

-(void) updateProgress:(double)size bytesDownloaded:(double) bytesDownloaded//VR:ProgressOption
{
	if (size <=0) {
		//	DDLogError("Progress not updated. Size is %f which is wrong.", size);
		return;
	}
	int progress = (bytesDownloaded/size) * 100;
	progress = progress < 100 ? progress : 100;//it could happen in the case of HLS
	self.progress.text = [NSString stringWithFormat:@"%d",progress];
}

@end

File: pcd/examples/VocSdkExampleSwift/VocSdkExampleSwift/VideoCell.swift
//
//  VideoCell.swift
//  VocSdkExampleSwift
//
//  Created by Gensch, Larry on 5/17/17.
//  Copyright © 2017 Akamai Technologies. All rights reserved.
//

import UIKit
import VocSdk

class VideoCell: UITableViewCell {

	@IBOutlet weak var title: UILabel!
	@IBOutlet weak var thumbnail: UIImageView!
	@IBOutlet weak var info: UILabel!
	@IBOutlet weak var info2: UILabel!
	@IBOutlet weak var progress: UILabel!

	override func awakeFromNib() {
        super.awakeFromNib()
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }

	func setBGColorAndStateForItem(item : VocItem?) {
		guard let item = item else { return }

		switch item.state {
		case .VOCItemDiscovered:
			self.backgroundColor = UIColor.brown
			self.info.text = "DISCOVERED"

		case .VOCItemIdle:
			self.backgroundColor = UIColor.orange
			self.info.text = "IDLE"

		case .VOCItemDownloading:
			self.backgroundColor = UIColor.gray
			self.info.text = "DISCOVERED"

		case .VOCItemQueued:
			self.backgroundColor = UIColor.gray
			self.info.text = "QUEUED"

		case .VOCItemCached :
			self.backgroundColor = UIColor.white
			self.info.text = "CACHED"

		case .VOCItemPartiallyCached:
			self.backgroundColor = UIColor.white
			self.info.text = "P-CACHED"

		case .VOCItemPaused:
			self.backgroundColor = UIColor.yellow
			self.info.text = "PAUSED"

		case .VOCItemFailed:
			self.backgroundColor = UIColor.red
			self.info.text = "FAILED"

		default:
			self.backgroundColor = UIColor.gray
		}
	}

	var _item : VocItem?

	/// The VocItem to display in the table cell
	var item : VocItem? {
		get { return _item }
		set {
			guard let item = newValue else { _item = nil; return }

			_item = newValue

			self.title.text = _item?.title

			if let thumbnail = item.thumbnail, let path = thumbnail.localPath {
				self.thumbnail.image = UIImage.init(contentsOfFile: path)
			} else {
				self.thumbnail.image = nil;
			}
			self.info2.text = "Content ID:\(String(describing: _item!.contentId))"

			self.setBGColorAndStateForItem(item: item)

			switch item.state {
			case .VOCItemCached, .VOCItemPartiallyCached:
				self.progress.isHidden = true

			default:
				self.progress.isHidden = false
			}
		}
	}

	func updateProgress(size: Double, bytesDownloaded: Double) -> Void {
		if (size <= 0) {
			print("Size is wrong: \(size)")
			return
		}
		var progress : Int = Int((bytesDownloaded * 100.00) / size * 100.00)
		progress = progress < 100 ? progress : 100
		self.progress.text = "\(progress)"
	}
}
